filament/backend/src/opengl/PlatformCocoaGL.mm

```diff
@@ -47,6 +53,83 @@
     void updateOpenGLContext(NSView *nsView, bool resetView, bool clearView);
 };
 
+CocoaGLSwapChain::CocoaGLSwapChain( NSView* inView )
+        : view(inView)
+        , previousBounds(NSZeroRect)
+        , previousWindowFrame(NSZeroRect)
+        , observers([NSMutableArray array])
+        , currentBounds(NSZeroRect)
+        , currentWindowFrame(NSZeroRect) {
+    NSView* __weak weakView = view;
+    NSMutableArray* __weak weakObservers = observers;
+    
+    void (^notificationHandler)(NSNotification *notification) = ^(NSNotification *notification) {
+        NSView* strongView = weakView;
+        if ((weakView != nil) && (weakObservers != nil)) {
+            this->currentBounds = [strongView convertRectToBacking: strongView.bounds];
+            this->currentWindowFrame = strongView.window.frame;
+        }
+    };
+    
+    // Various methods below should only be called from the main thread:
+    // -[NSView bounds], -[NSView convertRectToBacking:], -[NSView window],
+    // -[NSWindow frame], -[NSView superview],
+    // -[NSView setPostsFrameChangedNotifications:],
+    // -[NSView setPostsBoundsChangedNotifications:]
+    dispatch_async(dispatch_get_main_queue(), ^(void) {
+        NSView* strongView = weakView;
+		NSMutableArray* strongObservers = weakObservers;
+        if ((weakView == nil) || (weakObservers == nil)) {
```

Comment:
Please fix indentation here and throughout the rest of the file (we use 4 spaces, not tabs).

Reply:
done!